filters/fadein/fadein.go

```diff
@@ -0,0 +1,267 @@
+	}
+
+	h = net.JoinHostPort(h, p)
+	return s, h, nil
+}
+
+func normalizeEndpoint(e string) (string, string, error) {
+	u, err := url.Parse(e)
+	if err != nil {
+		return "", "", err
+	}
+
+	return normalizeSchemeHost(u.Scheme, u.Host)
+}
+
+func endpointKey(scheme, host string) string {
+	return fmt.Sprintf("%s://%s", scheme, host)
+}
+
+// endpointCreated("", <int,float64,string,time>)
+// endpointCreated("http://10.2.4.123:2156", $time)
+func (endpointCreated) CreateFilter(args []interface{}) (filters.Filter, error) {
+	if len(args) != 2 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	e, ok := args[0].(string)
+	if !ok {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	s, h, err := normalizeEndpoint(e)
+	if err != nil {
+		return nil, err
+	}
+
+	ec := endpointCreated{which: endpointKey(s, h)}
+	switch v := args[1].(type) {
+	case int:
+		ec.when = time.Unix(int64(v), 0)
+	case float64:
+		ec.when = time.Unix(int64(v), 0)
+	case string:
+		t, err := time.Parse(time.RFC3339, v)
+		if err != nil {
+			return nil, err
+		}
+
+		ec.when = t
+	case time.Time:
+		ec.when = v
+	default:
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	// mitigate potential flakyness caused by clock skew. When the created time is in the future based on
+	// the local clock, we ignore it.
+	now := time.Now()
+	if ec.when.After(now) {
+		log.Errorf("Endpoint created time in the future: %v. Potential clock skew.", ec.when)
+		ec.when = time.Time{}
+	}
+
+	return ec, nil
+}
+
+func (endpointCreated) Request(filters.FilterContext)  {}
+func (endpointCreated) Response(filters.FilterContext) {}
+
+// NewPostProcessor creates post-processor for maintaining the detection time of LB endpoints with fade-in
+// behavior.
+func NewPostProcessor() routing.PostProcessor {
+	return &postProcessor{
+		detected: make(map[string]detectedFadeIn),
+	}
+}
+
+func (p *postProcessor) Do(r []*routing.Route) []*routing.Route {
+	const configErrFmt = "Error while processing endpoint fade-in settings: %s, %s, %v."
+	now := time.Now()
+
+	for _, ri := range r {
+		if ri.Route.BackendType != eskip.LBBackend {
+			continue
+		}
+
+		ri.LBFadeInDuration = 0
+		ri.LBFadeInExponent = 1
+		endpointsCreated := make(map[string]time.Time)
+		for _, f := range ri.Filters {
+			if d, ok := f.Filter.(duration); ok {
+				ri.LBFadeInDuration = time.Duration(d)
+			}
+
+			if e, ok := f.Filter.(exponent); ok {
+				ri.LBFadeInExponent = float64(e)
+			}
+
+			if ec, ok := f.Filter.(endpointCreated); ok {
+				endpointsCreated[ec.which] = ec.when
+			}
+		}
+
+		if ri.LBFadeInDuration <= 0 {
+			continue
+		}
+
+		for i := range ri.LBEndpoints {
+			ep := &ri.LBEndpoints[i]
+
+			s, h, err := normalizeSchemeHost(ep.Scheme, ep.Host)
+			if err != nil {
+				log.Errorf(configErrFmt, ep.Scheme, ep.Host, err)
+				continue
+			}
+
+			key := endpointKey(s, h)
+			detected := p.detected[key].when
+			if detected.IsZero() || endpointsCreated[key].After(detected) {
+				detected = now
```

Comment:
No, that's not the purpose of the endpoint created value. The endpoints created value is used simply to detect when a new endpoint has the same hostname as an old one, or it was restarted. The reason why using it instead of 'now' would be bad is that there can be a large time between the created value and when we actually start sending traffic, and also an external API may send wrong values, too, e.g. due to bugs, while have full control over the internal timestamp.

Reply:
I see, thanks for clarifying.